src/main/Extensions/Base64Conversion/Types.ts

```diff
@@ -0,0 +1,9 @@
+export type InvocationArgument = {
+    invocationString: string;
+    encodingType: EncodingType;
+};
+
+export enum EncodingType {
+    Encode,
+    Decode,
+}
```

Comment:
I generally suggest avoiding "type files" where you store all your types.

Please create a dedicated file for each type. Also I suggest avoiding enums, just use a strings:

```typescript
export type EncodingType = "encode" | "decode";
```

Reply:
I was inspired by the types file from the Deepl Translator [Type](https://github.com/oliverschwendener/ueli/blob/main/src/main/Extensions/DeeplTranslator/Types.ts). 
Out of curiosity why strings over enums?
I've updated the code according to your suggestions, thank you for taking a look.

Reply:
I prefer string literal types over enums for the following reason, let's make a quick example:

### Example 1: Enum

Imagine you have this function definition that takes an enum as an argument:

```typescript
export enum Action {
    encode,
    decode,
}
```

```typescript
import { Action } from "./Action";

export const invoke(action: Action) => {
  // something happens here
};
```

If you want to invoke this function from somewhere else you also have to import the enum type, for example like this:

```typescript
import { Action } from "./Action";
import { invoke } from "./invoke";

invoke(Action.encode);
invoke(Action.decode);
```

In order to pass the right argument to the function you have to import an additional type.

### Example 2: String literal type

If you implement the same function with a string literal type, the additional type import is not needed:

```typescript
export const invoke(action: "encode" | "decode") => {
  // something happens here
};
```

```typescript
import { invoke } from "./invoke";

invoke("encode");
invoke("decode");
```

This is less code and also easier to use in my opinion.

Additionally, the electron API is also full of string literal types instead of enums and I like to stay consistent:

For example:

```typescript
browserWindow.setBackgroundMaterial("mica");

// instead of:
// browserWindow.setBackgroundMaterial(BackgroundMaterial.Mica)
```

But I guess this is because they want to be compatible with vanilla JavaScript which (I think) doesn't have enums.

I hope this helps :)